examples/java/src/main/java/org/apache/beam/examples/complete/cdap/PluginWrapper.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.examples.complete.cdap;
+
+/**
+ * Class wrapper for a plugin.
+ */
+public abstract class PluginWrapper<F, FP> {
```

Comment:
I think we should add bounds for generic types (e.x. `F extends ?`)

Reply:
Agree, it would be good, but this is not possible with the current solution. `F` may be `InputFormat<K, V>` or `OutputFormat<K, V>`. They don't extend common class or implement common interface. And they are placed in the dependency `org.apache.hadoop.mapreduce` so I can't put interface or class above them.